### PR TITLE
Don't expose internal model on the maker API

### PR DESCRIPTION
**Pull request by da-kami**
We don't have tests for the maker's API to create new offers.
We should not expose internal types that can change, but treat the HTTP API as a separate interface, otherwise we might accidentally break the API.

I really think we should stop exposing our internal model to the outside and keep proper API boundaries. 
Some of these changes might be controversion, feel free to come up with other ideas if you have better ones.

### CODE REVIEW

File: daemon/src/routes_maker.rs [hunk 1]
```diff
@@ -112,14 +111,29 @@ pub async fn post_sell_order(
     maker: &State<Maker>,
     _auth: Authenticated,
 ) -> Result<(), HttpApiProblem> {
+    let funding_rate = match order.funding_rate {
+        None => None,
+        Some(funding_rate) => Some(FundingRate::new(funding_rate).map_err(|e| {
+            HttpApiProblem::new(StatusCode::BAD_REQUEST)
+                .title("Invalid funding rate")
+                .detail(format!("{e:#}"))
+        })?),
```
Comment (thomaseizinger) [inline daemon/src/routes_maker.rs]: Can we do the expression with the `?` on a new line please? This took a while to parse in my head :D

**Review by thomaseizinger — APPROVED** [review]
Makes sense to me, we should check that the constructors for our newtypes enforce a certain precision (if we want that).

**Comment by da-kami** [comment]
Blocked until the new version is out because it is a breaking API change for the automation. Trivial to change but let's wait nonetheless.

**Review by luckysori — APPROVED** [review]

**Comment by thomaseizinger** [comment]
> Blocked until the new version is out because it is a breaking API change for the automation.

What part of this is a breaking change?

**Comment by da-kami** [comment]
Closing this because the API changed.

**Pull request closed without merging**